src/providers/getProvider.ts

```diff
@@ -1,28 +1,25 @@
-import config from 'config';
-import httpStatus from 'http-status-codes';
 import { container } from 'tsyringe';
-import { AppError } from '../common/appError';
-import { ProviderConfig } from '../common/interfaces';
+import { Logger } from '@map-colonies/js-logger';
+import { ProviderConfig, ProviderManager, ProvidersConfig } from '../common/interfaces';
+import { QueueFileHandler } from '../handlers/queueFileHandler';
+import { SERVICES } from '../common/constants';
 import { NFSProvider } from './nfsProvider';
 import { S3Provider } from './s3Provider';
 
-function getProvider(provider: string): S3Provider | NFSProvider {
-  switch (provider.toLowerCase()) {
-    case 'nfs':
-      return container.resolve(NFSProvider);
-    case 's3':
-      return container.resolve(S3Provider);
-    default:
-      throw new AppError(httpStatus.INTERNAL_SERVER_ERROR, `Invalid config provider received: ${provider} - available values:  "nfs" or "s3"`, false);
+function getProvider(config: ProviderConfig): S3Provider | NFSProvider {
+  const queueFileHandler: QueueFileHandler = container.resolve(SERVICES.QUEUE_FILE_HANDLER);
```

Comment:
Keep the master structure. use container.resolve for the entire class and not for every property

Reply:
ok